### PR TITLE
[EIP-712] Add eth_signTypedData_v4 rpc method

### PR DESCRIPTION
**Pull request by pilagod**
> This PR is stacked on #200 

This PR first handles `eth_signTypedData_v4` at provider and tests level. Upcoming PR will handle it at app side.

* Add `eth_signTypedData_v4` rpc method
* Refine account interface to support multiple formats of signature

### CODE REVIEW

File: packages/waallet/background/pool/request/handler.ts [hunk 1]
```diff
@@ -1,32 +1,41 @@
...
+    const requestId = uuidv4()
+    this.pool[requestId] = promise
     return requestId
   }
 
   public async wait(requestId: string) {
-    const result = await this.handleWait(requestId)
-    delete this.request[requestId]
+    const result = await this.pool[requestId]
+    delete this.pool[requestId]
```
Comment (WIN-LIN) [inline packages/waallet/background/pool/request/handler.ts:38]: I like this refactoring! it is more readable!

File: packages/waallet/background/pool/request/handler.ts [hunk 2]
```diff
@@ -111,17 +118,22 @@ export class TransactionRequestHandler implements RequestPool {
...
-    return this.pool[txId]
+/* EIP-712 Request Handler */
+
+export class Eip712RequestHandler {
+  public constructor(private accountManager: AccountManager) {}
+
+  public async send(data: { request: Eip712Request; accountId: string }) {
+    const { request, accountId } = data
+    const { account } = await this.accountManager.get(accountId)
+    return account.sign(request.hash(), SignatureFormat.Raw)
```
[the remark this answers is not in the document]
Comment (pilagod) [reply packages/waallet/background/pool/request/handler.ts:137]: I think `TypedDataEncoder` from `ethers` would check for us 🙈

**Review by WIN-LIN — APPROVED** [review]
LGTM, thanks!

**Comment by oneleo** [comment]
Note:
1. PasskeyAccount:
    Remove Eip191 from signRaw() (let AccountSkeleton determine the signature type).

2. SimpleAccount:
    Change signRaw()'s signMessage() to sign() to remove Eip191 (let AccountSkeleton determine the signature type).

3. AccountSkeleton:
    (1) Update signRaw() (formerly sign()) to not require metadata.
    (2) Add a sign() method to sign based on type (Eip191 or Raw).

4. Provider’s `handleSendTransaction()` and `handleSignTypedDataV4()`:
    Location: packages/waallet/background/provider.ts

    1. `RequestHandler` is an instance of `RequestPool interface`.
        a. Include `TransactionRequestHandler` and `Eip712RequestHandler` when constructing `RequestHandler`.
        b. TransactionRequestHandler's `send()` uses Eip191.
        c. Eip712RequestHandler's `send()` uses Raw.
    2. `TransactionRequest` is an instance of `Request interface` and carries "transaction" parameters.
    3. `Eip712Request` is an instance of `Request interface` and carries "eip712" parameters.
    4. RequestHandler's `send()` calls either TransactionRequestHandler’s or Eip712RequestHandler's `send()` based on the `Request` type.
    5. `RequestPool interface` includes node and bundler; `Request interface` includes request (tx or eip712) information.

**Review by oneleo — APPROVED** [review]
Passed: 
ETH and Token transfers using PasskeyAccount v0.7 and SimpleAccount v0.7.

**Pull request merged**